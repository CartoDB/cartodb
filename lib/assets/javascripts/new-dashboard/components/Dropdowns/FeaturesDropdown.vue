<template>
  <div class="features-dropdown">
    <slot />
    <div class="dropdown-container">
      <ul class="list">
        <li class="element" v-for="element in list" :key="element.name || element">
          <a :href="element.url" class="list-text text is-small" v-if="element.url">{{ element.name }}</a>

          <router-link :to="{ name: linkRoute, params: routeParams(element) }" class="list-text text is-small" v-if="!element.url">
            {{ element }}
          </router-link>
        </li>

        <li class="element" v-if="$slots.footer" @click.stop.prevent="noop">
          <span class="list-text text is-small footer">
            <slot name="footer" />
          </span>
        </li>
      </ul>
    </div>
  </div>
</template>

<script>
export default {
  name: 'FeaturesDropdown',
  props: {
    list: Array,
    feature: String,
    linkRoute: String
  },
  methods: {
    routeParams (element) {
      return { [this.feature]: element };
    },
    noop () {}
  }
};
</script>

<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.features-dropdown {
  &:hover {
    .dropdown-container {
      opacity: 1;
      pointer-events: all;
    }

    .feature-text {
      color: $primary-color;
    }
  }
}

.list-text {
  display: inline-block;
  width: 100%;
  padding: 8px 22px;
}

.dropdown-container {
  position: absolute;
  z-index: 1;
  min-width: 60%;
  padding-top: 16px;
  transform: translateX(-12px);
  transform: all 0.3s linear;
  opacity: 0;
  pointer-events: none;
}

.list {
  position: relative;
  padding: 0;
  border: 1px solid $light-grey;
  border-radius: 4px;
  background-color: $white;

  &::after {
    content: '';
    position: absolute;
    top: -9px;
    left: 24px;
    width: 16px;
    height: 16px;
    transform: rotate(45deg);
    border: 1px solid $light-grey;
    border-right: none;
    border-bottom: none;
    border-radius: 2px;
    background-color: $white;
  }
}

.element {
  &:first-of-type {
    padding-top: 8px;
  }

  &:last-of-type {
    padding-bottom: 8px;
  }
}

.footer {
  color: $text-color;
  cursor: default;
}
</style>
