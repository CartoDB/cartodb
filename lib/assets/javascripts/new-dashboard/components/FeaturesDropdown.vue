<template>
  <div class="features-dropdown">
    <span class="feature-text text is-caption is-txtGrey">{{list.length}} {{feature}}</span>
    <div class="dropdown-container">
      <ul class="list">
        <li class="element" v-for="elem in list" :key="elem">
          <a href="#" class="list-text text is-small">{{elem}}</a>
        </li>
      </ul>
    </div>
  </div>
</template>

<script>

export default {
  name: 'FeaturesDropdown',
  props: {
    list: Array,
    feature: String
  }
};
</script>

<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.features-dropdown {
  &:hover {
    .dropdown-container {
      opacity: 1;
      pointer-events: all;
    }

    .feature-text {
      color: $primary-color;
    }
  }
}

.list-text {
  display: inline-block;
  width: 100%;
  padding: 8px 22px;
}

.dropdown-container {
  position: absolute;
  z-index: 1;
  min-width: 60%;
  padding-top: 16px;
  transform: translateX(-12px);
  transform: all 0.3s linear;
  opacity: 0;
  pointer-events: none;
}

.list {
  position: relative;
  padding: 0;
  border: 1px solid $light-grey;
  border-radius: 4px;
  background-color: $white;

  &::after {
    content: '';
    position: absolute;
    top: -9px;
    left: 24px;
    width: 16px;
    height: 16px;
    transform: rotate(45deg);
    border: 1px solid $light-grey;
    border-right: none;
    border-bottom: none;
    border-radius: 2px;
    background-color: $white;
  }
}

.element {
  &:first-of-type {
    padding-top: 8px;
  }

  &:last-of-type {
    padding-bottom: 8px;
  }
}
</style>
