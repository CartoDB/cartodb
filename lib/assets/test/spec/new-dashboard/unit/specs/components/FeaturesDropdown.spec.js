import { shallowMount } from '@vue/test-utils';
import FeaturesDropdown from 'new-dashboard/components/FeaturesDropdown';

describe('FeaturesDropdown.vue', () => {
  it('should render correct contents', () => {
    const featuresDropdown = shallowMount(FeaturesDropdown, {
      propsData: {
        list: ['one tag test'],
        feature: 'tag'
      }
    });
    expect(featuresDropdown).toMatchSnapshot();
  });

  it('should render correct multiple elements', () => {
    const featuresDropdown = shallowMount(FeaturesDropdown, {
      propsData: {
        list: ['my first map', 'another map', 'Madrid visualization', 'NoSpacesMap'],
        feature: 'maps'
      }
    });
    expect(featuresDropdown).toMatchSnapshot();
  });
});
