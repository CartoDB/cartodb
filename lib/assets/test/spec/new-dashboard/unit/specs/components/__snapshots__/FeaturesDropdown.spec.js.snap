// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturesDropdown.vue should render correct contents 1`] = `
<div class="features-dropdown"><span class="feature-text text is-caption is-txtGrey">1 tag</span>
  <div class="dropdown-container">
    <ul class="list">
      <li class="element"><a href="#" class="list-text text is-small">one tag test</a></li>
    </ul>
  </div>
</div>
`;

exports[`FeaturesDropdown.vue should render correct multiple elements 1`] = `
<div class="features-dropdown"><span class="feature-text text is-caption is-txtGrey">4 maps</span>
  <div class="dropdown-container">
    <ul class="list">
      <li class="element"><a href="#" class="list-text text is-small">my first map</a></li>
      <li class="element"><a href="#" class="list-text text is-small">another map</a></li>
      <li class="element"><a href="#" class="list-text text is-small">Madrid visualization</a></li>
      <li class="element"><a href="#" class="list-text text is-small">NoSpacesMap</a></li>
    </ul>
  </div>
</div>
`;
